go/signedexchange/certurl/parse.go

```diff
@@ -0,0 +1,105 @@
+package certurl
+
+import (
+	"bytes"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+)
+
+func writeHead(w *bytes.Buffer, n int, size int) error {
+	// encoding/binary's Write is not available here
+	// since the size is not necessarily a power of 2.
+	for i := 0; i < size; i++ {
+		if err := w.WriteByte(byte(n >> (8 * uint(size-i-1)))); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// CertificateMessageFromPEM parses a PEM formatted content to a certUrl content.
+// See https://wicg.github.io/webpackage/draft-yasskin-http-origin-signed-responses.html for the spec.
+func CertificateMessageFromPEM(pemFileContent []byte) ([]byte, error) {
+	b := pemFileContent
+
+	entries := []*x509.Certificate{}
+	totalLength := 0
+	for {
+		block, rest := pem.Decode(b)
+		if block == nil && len(rest) > 0 {
+			return nil, fmt.Errorf("failed to parse PEM file")
+		}
+		c, err := x509.ParseCertificate(block.Bytes)
+		if err != nil {
+			return nil, err
+		}
+
+		entries = append(entries, c)
+		totalLength += len(c.Raw)
+
+		if len(rest) == 0 {
+			break
+		}
+		b = rest
+	}
+
+	buf := &bytes.Buffer{}
+
+	// enum {
+	//     X509(0),
+	//     RawPublicKey(2),
+	//     (255)
+	// } CertificateType;
+	//
+	// struct {
+	//     select (certificate_type) {
+	//     case RawPublicKey:
+	//         /* From RFC 7250 ASN.1_subjectPublicKeyInfo */
+	//         opaque ASN1_subjectPublicKeyInfo<1..2^24-1>;
+	//
+	//     case X509:
+	//         opaque cert_data<1..2^24-1>;
+	//     };
+	//     Extension extensions<0..2^16-1>;
+	// } CertificateEntry;
+	//
+	// struct {
+	//     opaque certificate_request_context<0..2^8-1>;
+	//     CertificateEntry certificate_list<0..2^24-1>;
+	// } Certificate;
```

Comment:
FYI, @sleevi tells me that using the TLS Certificate format is likely to cause problems for implementers, since its parser isn't actually exposed by TLS validation libraries. He suggests I just do a CBOR structure holding an array of certificates, an ocsp response, and an sct response. That's also easier for you to test, so don't spend too much time getting the Certificate message right in this change.